src/main/java/com/puppycrawl/tools/checkstyle/xpath/XpathQueryGenerator.java

```diff
@@ -69,24 +73,54 @@
  * @author Timur Tibeyev.
  */
 public class XpathQueryGenerator {
+    /** Default distance between tab stops. */
+    private static final int DEFAULT_TAB_WIDTH = 4;
     /** The root ast. */
     private final DetailAST rootAst;
     /** The line number of the element for which the query should be generated. */
     private final int lineNumber;
     /** The column number of the element for which the query should be generated. */
     private final int columnNumber;
+    /** The file name. */
+    private final FileText fileText;
+    /** The distance between tab stop position. */
+    private final int tabWidth;
 
     /**
      * Creates a new {@code XpathQueryGenerator} instance.
      *
      * @param rootAst root ast
      * @param lineNumber line number of the element for which the query should be generated
      * @param columnNumber column number of the element for which the query should be generated
+     * @param fileName path to the file
+     * @param tabWidth distance between tab stop position
      */
-    public XpathQueryGenerator(DetailAST rootAst, int lineNumber, int columnNumber) {
+    public XpathQueryGenerator(DetailAST rootAst, int lineNumber, int columnNumber,
+                               String fileName, int tabWidth) {
         this.rootAst = rootAst;
         this.lineNumber = lineNumber;
         this.columnNumber = columnNumber;
+        try {
+            fileText = new FileText(new File(fileName),
+                    System.getProperty("file.encoding", "UTF-8"));
```

Comment:
Use StandardCharset.UTF-8

Reply:
@MEZk , does it make sense to forbid such string by http://checkstyle.sourceforge.net/config_coding.html#IllegalTokenText ?

Reply:
@romani 
Yes, I noticed that many contributors do not know StandardCharsets.
Created the issue https://github.com/checkstyle/checkstyle/issues/5023